lib/player.rb

```diff
@@ -0,0 +1,95 @@
+require_relative 'scoring'
+require_relative 'tile_bag'
+
+module Scrabble
+
+  class Player
+
+    attr_reader :name, :plays, :total_score, :tiles
+
+    def initialize(name)
+      @name = name
+      @plays = []
+      @total_score = 0
+      @tiles = []
+    end
+
+    def string?(word)
+      return word.class == String
+    end #end of string?
+
+    def valid?(word)
+      count = 0
+      length = 0
+      for i in 0 ... word.length
+        if @tiles.include?(word[i].upcase.to_sym)
+          count += 1
+          length += 1
+        end
+      end
+      return length == word.length
+    end #end of valid?
+
+
+    def play(word)
+      if won?
+        return false
+      end
+
+      if  string?(word) && valid?(word)
+
+        score = Scoring.score(word)
+        @total_score += score
+
+        index = 0
+        while index < @tiles.length
+          if word.upcase.include?(@tiles[index].to_s)
+            @tiles.delete(@tiles[index])
+          else
+            index += 1
+          end
```

Comment:
This `while` loop code is incorrect. Here is a test case that should expose the issue:

```ruby
it "should remove only the tiles needed to play the given word" do
  # WARNING: This is a hacky thing to force a particular value for the @tiles variable.
  # We would not normally do this as there are much better ways to accomplish the
  # same thing in our tests -- they're just too much code to put into this comment.
  @person.instance_variable_set(:@tiles, ["C", "H", "E", "E", "S", "E", "S"])

  @person.play("CHESS")

  @person.tiles.must_equal ["E", "E"]
end
```